tests/utils/console_test.py

```diff
@@ -0,0 +1,162 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import random
+import pytest
+import unittest.mock as mock
+
+from unittest import TestCase
+from esrally.utils import console
+
+
+class ConsoleFunctionTests(TestCase):
+    @mock.patch("os.environ.get")
+    def test_global_rally_running_in_docker_is_false(self, patched_env_get):
+        patched_env_get.return_value = random.choice(["false", "False", "FALSE"])
+
+        _ = console.init()
+        self.assertEqual(False, console.RALLY_RUNNING_IN_DOCKER)
+
+    @mock.patch("os.environ.get")
+    def test_global_rally_running_in_docker_is_false_if_unset(self, patched_env_get):
+        patched_env_get.return_value = ""
+
+        _ = console.init()
+        self.assertEqual(False, console.RALLY_RUNNING_IN_DOCKER)
+
+    @mock.patch("os.environ.get")
+    def test_global_rally_running_in_docker_is_true(self, patched_env_get):
+        patched_env_get.return_value = random.choice(["true", "True", "TRUE"])
+
+        _ = console.init()
+        self.assertEqual(True, console.RALLY_RUNNING_IN_DOCKER)
+
+    @mock.patch("sys.stdout.isatty")
+    @mock.patch("builtins.print")
```

Comment:
I wonder if we should define defaults in `console` and allow to override them instead of monkey-patching? E.g. allow to pass the `print` implementation or just have a global variable and change it in the test?